comp/serializer/logscompression/def/component.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package logscompression provides the component for logs compression
+package logscompression
+
+// team: agent-processing-and-routing
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/compression"
+)
+
+// Component is the component type.
+type Component interface {
+	NewCompressor(kind string, level int) compression.Compressor
```

Comment:
Can you explain what are the reason that  `logscompression` is a factory unlike `metricscompression`?